src/vstest.console/TestPlatformHelpers/TestRequestManager.cs

```diff
@@ -125,8 +129,14 @@ public bool DiscoverTests(DiscoveryRequestPayload discoveryPayload, ITestDiscove
 
             bool success = false;
 
+            var runsettings = discoveryPayload.RunSettings;
+            if (this.TryUpdateDesignMode(runsettings, out string updatedRunsettings))
```

Comment:
We've the right fix in master already, it requires change in ArgumentProcessorPriorities. Brought in only the minimal required for RTW; no loggers require designmode for RTW.